Classes/Controller/ClipboardStateContainer.php

```diff
@@ -0,0 +1,80 @@
+<?php
+namespace Neos\Neos\Ui\Controller;
+
+/*
+ * This file is part of the Neos.Neos.Ui package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Error\Messages\Message;
+use Neos\ContentRepository\Domain\Model\NodeInterface;
+
+/**
+ * This is a container for clipboard state that needs to be persisted server side
+ *
+ * @Flow\Scope("session")
+ */
+class ClipboardStateContainer
+{
+    /**
+     * @var NodeInterface
+     */
+    protected $clipboardSubject = null;
```

Comment:
why `protected`?
```
/**
 * @var NodeInterface
 */
private $nodeInClipboard = null;
```

Reply:
Because the first thing I heard when meeting the Neos project is to never use `private`. I think the reasoning was that you might want to extend the class later. Frankly I never personally agreed with it, but just got used to doing it to follow the supposed convention. But if it's no longer the case I'd gladly do that.

Reply:
Oh gosh, it has to be protected in order for it to be persisted in the session! Waisted so much time debugging this...